ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/views/HivePushdownSnapshotFilterRule.java

```diff
@@ -116,15 +116,16 @@ private boolean setSnapShotId(RexNode op1, RexNode op2) {
         return false;
       }
 
-      long snapshotId = literal.getValueAs(Long.class);
+      Long snapshotId = literal.getValueAs(Long.class);
 
       RelOptTable relOptTable = getRelOptTableOf(op2);
       if (relOptTable == null) {
         return false;
       }
 
       RelOptHiveTable hiveTable = (RelOptHiveTable) relOptTable;
-      hiveTable.getHiveTableMD().setVersionIntervalFrom(Long.toString(snapshotId));
+      String snapshotIdText = snapshotId != null ? Long.toString(snapshotId) : null;
```

Comment:
````
hiveTable.getHiveTableMD().setVersionIntervalFrom(String.valueOf(snapshotId))
````

Reply:
`String.valueOf(null)` returns string literal `"null"`. This is not a valid snapshotId

Reply:
Objects.toString(obj, null)

Reply:
refactored